### PR TITLE
Add Native Source Editor - Part 1

### PR DESCRIPTION
**Pull request by tonisevener**
**Phabricator:** 
https://phabricator.wikimedia.org/T331936

### Notes
This PR adds the native editor feature flag and implements the initial pieces of it within `PageEditorViewController`. I also added a full source edit button in `ArticleViewController`, so that we can test performance against a large set of wikitext as we build this. The entry point of this full source edit button is likely to change before release, though.

Note I found a bug in the default automatic theme switching in this branch. Marking as draft while I look into that.

### Test Steps
1. In the Staging scheme, load the article editor, either by tapping the full source "Edit" button at the top of an article or any edit pencil. 
2. Text formatting toolbars and input views should display like our Code Mirror editor, though state logic in the Find and Replace toolbar view will act wrong. I will fix this in a separate find & replace PR later on.

### CODE REVIEW

**Comment by tonisevener** [comment]
> Note I found a bug in the default automatic theme switching in this branch. Marking as draft while I look into that.

I can reproduce this bug in `main` as well, so it's unrelated to these changes. Marking as ready to review.

File: Wikipedia/Code/WMFAppViewController.m [hunk 1]
```diff
@@ -1853,6 +1854,7 @@ - (void)updateAppThemeIfNecessary {
     if (self.theme != theme) {
         [self applyTheme:theme];
         [self.settingsViewController loadSections];
+        [self setWKAppEnvironmentThemeWithTheme: theme];
```
Comment (staykids) [inline Wikipedia/Code/WMFAppViewController.m:1857]: Just a note for forthcoming PRs that we'd eventually need to also pass along the trait collection changes so the font scaling respects our design intents.

File: Wikipedia/Code/PageEditorViewController.swift [hunk 2]
```diff
@@ -0,0 +1,236 @@
+import UIKit
+import Components
+
+protocol PageEditorViewControllerDelegate: AnyObject {
+    func pageEditorDidCancelEditing(_ pageEditor: PageEditorViewController, navigateToURL: URL?)
+}
+
+class PageEditorViewController: UIViewController {
```
Comment (staykids) [inline Wikipedia/Code/PageEditorViewController.swift:8]: Should we mark this as `final`?

File: Wikipedia/Code/PageEditorViewController.swift [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        
+        navigationItemController.apply(theme: theme)
+        view.backgroundColor = theme.colors.paperBackground
+    }
+}
+
+// MARK: - WKSourceEditorViewControllerDelegate
+
+extension PageEditorViewController: WKSourceEditorViewControllerDelegate {
+    func sourceEditorViewControllerDidTapFind(sourceEditorViewController: Components.WKSourceEditorViewController) {
```
Comment (staykids) [inline Wikipedia/Code/PageEditorViewController.swift:160]: The `Components.` namespace prefix is unneeded here.

**Review by staykids — COMMENTED** [review]
Soooo cool seeing this come together! I ignored text theming stuff as I assume it's coming in forthcoming PRs. Just a couple of minor notes in `PageEditorViewController`, but happy to merge as-is.

File: Wikipedia/Code/WMFAppViewController.m (hunk 1 above)
Comment (tonisevener) [reply Wikipedia/Code/WMFAppViewController.m]: Thanks - this is updated!

File: Wikipedia/Code/PageEditorViewController.swift (hunk 2 above)
Comment (tonisevener) [reply Wikipedia/Code/PageEditorViewController.swift]: Agreed, done!

File: Wikipedia/Code/PageEditorViewController.swift (hunk 3 above)
Comment (tonisevener) [reply Wikipedia/Code/PageEditorViewController.swift]: Done!

File: Wikipedia/Code/WMFAppViewController.m (hunk 1 above)
Comment (JoeWarwick) [reply Wikipedia/Code/WMFAppViewController.m]: good stuff

**Review by mazevedofs — APPROVED** [review]
looking great!

**Pull request merged**